### PR TITLE
docker_container_exec: allow to specify environment

### LINKED ISSUE #248: Support --env in docker_container_exec
##### SUMMARY
`docker exec` supports an `--env` argument (see https://docs.docker.com/engine/reference/commandline/exec).
Would be nice if the module docker_container_exec would support this too.

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
docker_container_exec

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Fixes #248.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
docker_container_exec

### CODE REVIEW

**Comment by felixfontein** [comment]
(Right now includes #253, will rebase once that's merged.)

File: plugins/modules/docker_container_exec.py [hunk 1]
```diff
@@ -183,6 +194,15 @@ def main():
     stdin = client.module.params['stdin']
     strip_empty_ends = client.module.params['strip_empty_ends']
     tty = client.module.params['tty']
+    env = client.module.params['env']
+
+    if env is not None:
+        for name, value in sorted(env.items()):
```
Comment (briantist) [inline plugins/modules/docker_container_exec.py:200]: not there's anything wrong with it, but just wondering why this is sorted?

**Review by briantist — APPROVED** [review]
Looks good to me! No blocking questions or concerns.

File: plugins/modules/docker_container_exec.py (hunk 1 above)
Comment (felixfontein) [reply plugins/modules/docker_container_exec.py:200]: I think I wanted to write `list`, not `sorted`. The result is the same, except that `sorted` is a bit slower... Will change it.

**Review by briantist — APPROVED** [review]

**Comment by felixfontein** [comment]
@briantist thanks for also reviewing this one!

**Pull request merged**